charms/argo-controller/src/charm.py

```diff
@@ -54,260 +80,93 @@ def __init__(self, *args):
-                        "imagePullPolicy": "Always",
-                        "args": [
-                            "--configmap",
-                            "argo-controller-configmap-config",
-                            "--executor-image",
-                            executor_image,
-                        ],
-                        "envConfig": {
-                            "ARGO_NAMESPACE": self.model.name,
-                            "LEADER_ELECTION_IDENTITY": self.model.app.name,
-                        },
-                        "volumeConfig": [
-                            {
-                                "name": "configmap",
-                                "mountPath": "/config-map.yaml",
-                                "files": [
-                                    {
-                                        "path": "config",
-                                        "content": yaml.dump(config_map),
-                                    }
-                                ],
-                            }
-                        ],
-                    },
-                ],
-                "kubernetesResources": {
-                    "customResourceDefinitions": [
-                        {"name": crd["metadata"]["name"], "spec": crd["spec"]} for crd in crds
-                    ],
-                    "secrets": [
-                        {
-                            "name": "mlpipeline-minio-artifact",
-                            "type": "Opaque",
-                            "data": {
-                                "accesskey": b64encode(os["access-key"].encode("utf-8")),
-                                "secretkey": b64encode(os["secret-key"].encode("utf-8")),
-                            },
-                        }
-                    ],
-                },
-            }
+                lightkube_client=lightkube.Client(),
+            ),
+            depends_on=[
+                self.leadership_gate,
+                self.object_storage_relation,
+            ],
         )
 
-        self.model.unit.status = ActiveStatus()
-
-    def _check_leader(self):
-        if not self.unit.is_leader():
-            # We can't do anything useful when not the leader, so do nothing.
-            raise CheckFailed("Waiting for leadership", WaitingStatus)
-
-    def _get_interfaces(self):
-        try:
-            interfaces = get_interfaces(self)
-        except NoVersionsListed as err:
-            raise CheckFailed(err, WaitingStatus)
-        except NoCompatibleVersions as err:
-            raise CheckFailed(err, BlockedStatus)
-        return interfaces
-
-    def _check_object_storage(self, interfaces):
-        if not ((os := interfaces["object-storage"]) and os.get_data()):
-            raise CheckFailed("Waiting for object-storage relation data", BlockedStatus)
-
-        return list(os.get_data().values())[0]
+        self.profile_controller_container = self.charm_reconciler.add(
```

Comment:
nit: the name shouldn't be `profile_controller_container` here

Reply:
Updated.